[5.5.x] Parsing error no longer tiggers failed probe

bernardjkim opened the pull request:
Parsing errors will only be logged and will no longer trigger a failed probe.

Resolves gravitational/gravity#1307

bernardjkim committed:
Parsing error does not affect status

r0mant reviewed (approved):
@bernardjkim Could you please also verify that clusters no longer enter degraded state due to this upon fresh install?

a-palchikov commented:
Why is there a parsing error in the first place - the referenced issue does not mention this?

bernardjkim commented:
@a-palchikov 
Good question, I added some additional information in the referenced issue. 

If you look through the received nethealth metrics, it does not contain any request/timeout counters. 
In order for nethealth to initialize its counters, it needs to first discover the other nethealth pods in the cluster, so there is some duration after initialization when nethealth metrics will return no counters.

This issue probably would have gone unnoticed since the initial discovery should run within ~10 seconds, but because I had removed the service resource from the monitoring-app, this initial discovery was unable to execute successfully.
